sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/MetricHelper.cs

```diff
@@ -14,22 +14,21 @@ internal class MetricHelper
 
         internal static List<TelemetryItem> OtelToAzureMonitorMetrics(Batch<Metric> batch, string roleName, string roleInstance, string instrumentationKey)
         {
-            List<TelemetryItem> telemetryItems = new List<TelemetryItem>();
-            TelemetryItem telemetryItem;
-
+            List<TelemetryItem> telemetryItems = new();
             foreach (var metric in batch)
             {
-                if (metric.MetricType == MetricType.DoubleSum || metric.MetricType == MetricType.DoubleGauge)
+                if (MetricsData.IsSupportedType(metric.MetricType))
                 {
                     foreach (ref readonly var metricPoint in metric.GetMetricPoints())
                     {
-                        telemetryItem = new TelemetryItem(metricPoint.EndTime.UtcDateTime, roleName, roleInstance, instrumentationKey);
-                        telemetryItem.Data = new MonitorBase
+                        telemetryItems.Add(new TelemetryItem(metricPoint.EndTime.UtcDateTime, roleName, roleInstance, instrumentationKey)
```

Comment:
while nothing wrong with this refactoring, i'd strongly suggest to keep it separate PR, to make it easy for reviewers to.